src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -570,18 +554,18 @@ private function _process_directives( string $html ) {
 	 * @since 6.6.0 The function now adds a warning when the namespace is null, falsy, or the directive value is empty.
 	 * @since 6.6.0 Removed `default_namespace` and `context` arguments.
 	 * @since 6.6.0 Add support for derived state.
+	 * @since 6.9.0 Recieve $entry as an argument instead of the directive value string.
 	 *
-	 * @param string|true $directive_value The directive attribute value string or `true` when it's a boolean attribute.
+	 * @param array $entry An array containing a whole directive entry with its namespace, value, suffix, or unique ID.
 	 * @return mixed|null The result of the evaluation. Null if the reference path doesn't exist or the namespace is falsy.
 	 */
-	private function evaluate( $directive_value ) {
-		$default_namespace = end( $this->namespace_stack );
-		$context           = end( $this->context_stack );
+	private function evaluate( $entry ) {
+		$context                               = end( $this->context_stack );
+		['namespace' => $ns, 'value' => $path] = $entry;
 
-		list( $ns, $path ) = $this->extract_directive_value( $directive_value, $default_namespace );
 		if ( ! $ns || ! $path ) {
 			/* translators: %s: The directive value referenced. */
-			$message = sprintf( __( 'Namespace or reference path cannot be empty. Directive value referenced: %s' ), $directive_value );
+			$message = sprintf( __( 'Namespace or reference path cannot be empty. Directive value referenced: %s' ), json_encode( $entry ) );
 			_doing_it_wrong( __METHOD__, $message, '6.6.0' );
```

Comment:
I was considering removing this `_doing_it_wrong` call, as mentioned here. For now, we can leave it there while we decide on a better solution to display errors and warnings during iAPI's directive processing.

I'd change the message, though, so we don't call `json_encode()`.

```suggestion
			/* translators: 1: Entry's namespace, 2: Entry's value. */
			$message = sprintf( __( 'Passed entry should contain a defined namespace and value. Namespace: %1$s. Value: %2$s' ), $ns, $path );
			_doing_it_wrong( __METHOD__, $message, '6.9.0' );
```

Reply:
I agree with removing this and focusing on reporting these kinds of errors in JavaScript instead of PHP.

Reply:
I've removed it in this commit: https://github.com/WordPress/wordpress-develop/pull/10284/commits/b37dc21e3cae3d9991132c3f8de5ca4c1a7cefc9